core/src/test/java/org/apache/accumulo/core/client/mapred/AccumuloMultiTableInputFormatTest.java

```diff
@@ -29,24 +29,20 @@
 import org.apache.accumulo.core.util.Pair;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapred.JobConf;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TestName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
 
 @Deprecated(since = "2.0.0")
 public class AccumuloMultiTableInputFormatTest {
 
-  @Rule
-  public TestName testName = new TestName();
-
   /**
    * Verify {@link org.apache.accumulo.core.client.mapreduce.InputTableConfig} objects get correctly
    * serialized in the JobContext.
    */
   @Test
-  public void testTableQueryConfigSerialization() {
-    String table1Name = testName.getMethodName() + "1";
-    String table2Name = testName.getMethodName() + "2";
+  public void testTableQueryConfigSerialization(TestInfo testInfo) {
+    String table1Name = testInfo.getDisplayName() + "1";
```

Comment:
The display name can contain invalid characters (commas) that aren't suitable for table names. It's better to use the `.getTestMethod().get()` (or some other variant that ensures it's not null, like `.getTestMethod().orElseThrow(IllegalStateException::new)`).

Reply:
Good point. I was looking for a good replacement for `.getMethodName()` and from some quick testing after your suggestions, it looks like `testInfo.getTestMethod().orElseThrow(IllegalStateException::new).getName()` might be a suitable replacement.

Reply:
That's a bit bulky to replace everywhere. I'd put a static method somewhere. Maybe in `AccumuloITBase`:

```java
  public static testMethodName(TestInfo testInfo) {
    return testInfo.getTestMethod().orElseThrow(IllegalStateException::new).getName();
  }
```

That way, you can do:

```java
import static org.apache.accumulo.harness.AccumuloITBase.testMethodName;
// ...
@Test
public void someTestName(TestInfo testInfo) {
  String table1Name = testMethodName(testInfo) + "1";
  // ...
}

```

Reply:
I like this idea. I think I'll forgo creating that method until I work on that module (accumulo-test, where AccumuloITBase resides) but I have left a note about creating the method and using it in the TODO section on ticket #2441

Reply:
I had an alternate idea for this, in my latest code review, using a `Function<TestInfo,String>` or similar. The suggestion above could also be done that way, but having it live in AccumuloITBase would only affect integration tests. The tests in other modules would still benefit from using something shorter in many of the tests. See my recent code review for an example.

Reply:
I found the best way to do it is to create a superclass for the test that has a `testName()` method and a `@BeforeEach` annotation that sets the test name from the `TestInfo` class. See `WithTestNames` in the core module. The same can be done for other modules that need it. However, the class would need to be duplicated for any modules that use it, because of the way test code does not have dependencies on test code in other modules, and because we can't put the test class with JUnit types into the main jar, or else it would add a JUnit dependency there, which we don't want.